applications/HDF5Application/python_scripts/core/file_io.py

```diff
@@ -80,24 +81,47 @@ def _GetIO(io_type):
 class _FilenameGetter(object):
 
     def __init__(self, settings):
-        filename = settings['file_name']
-        self.filename_parts = filename.split('<time>')
+        self.filename = settings['file_name']
+        self.model_part = None
+
+        if (self.filename.find("<rank>") != -1):
+            raise Exception("Flag \"<rank>\" is not allowed to be used in HDF5 output file namings. Please remove it [ filename = \"" + self.filename + "\" ].")
+
+        if (self.filename.find("<step>") != -1):
```

Comment:
why not?

Reply:
The time step is required to generate xdmf files properly to visualize in paraview. That is the only reason to avoid having `<step>` flag. (If some choses `<step>` flag and not include the `<time>` flag, he will generate correctly the output h5 files having seperate file for each time step, but will have problems in generating proper xdmf files afterwards. I thought of not to allow `<step>` so user is forced to use `<time>` to have seperate time step files).

Reply:
can you explain more in detail?
I am asking bcs the vtk has the exact opposite problem

Reply:
When xdmf files are created, in the xdmf file hdf5 files are linked for each step. Each step requires the timestep value, which is taken from the filename. If user provides only `<step>` flag then, when xdmf files are created, there is no way to know about the time step value. (HDF5 application doesnot open time step results h5 files when creating xdmf). But if the user is not allowed to use `<step>` then user will be forced to use `<time>` flag in the filename which will be used in xdmf file time step linking.

This has nothing to do with `FileNameDataCollector`. `FileNameDaataCollector` supports properly all `<rank>`, `<step>` and `<time>`. This is only a limitation of h5 files.

Reply:
Lets speak about this on the phone, I think it will be more productive than writing many messages :)

Reply:
done :)

Reply:
done :)